MANUAL: Show for which formats an extension is supported on hover

not-my-profile opened the pull request:
This commit introduces a new extension-support.lua Lua filter
for MANUAL.html that recognizes headings like the following:

#### Extension: `fenced_code_blocks` ####

And appends a Span with content 🗹 to the heading.
The Span has a title attribute (which browsers display on hover)
with automatically generated text like the following:

  enabled by default for:
  * ipynb
  * markdown
  * markdown_github
  * markdown_phpextra
  * opml

  disabled by default for:
  * markdown_mmd
  * markdown_strict
  * plain

Letting readers quickly check for which formats a specific extension
is supported / enabled by default / disabled by default.

jgm commented:
This is clever, but I'm not sold on it.
It seems to me that what is useful is knowing which extensions are enabled/disabled in a given format which you're planning to use.  Knowing which formats are enabled by default for a given extension seems less important.

not-my-profile commented:
>Knowing which formats are enabled by default for a given extension seems less important.

I find it primarily useful for checking if a Markdown extension is supported for CommonMark.

not-my-profile commented:
When reading about a feature in the MANUAL a logical question is which formats do actually support that feature.

E.g. many Markdown extensions are not supported by CommonMark.

not-my-profile commented:
Well I think it's a shame that you don't see the appeal. @tarleb what do you think about this?

tarleb commented:
Having this information in the manual seems like a good thing, I'm only worried that the manual becomes even more overwhelming. Can you deploy an example build somewhere for us to look at?

not-my-profile commented:
Sure: https://share.push-f.com/pandoc/MANUAL.html#other-extensions

tarleb commented:
Pros:

- interesting, easy to access info
- compact
- more up-to-date than some of the documentation (e.g., commonmark_x is missing from the lists in the docs for `smart`)

Cons:

- doesn't work on mobile (at least for me)
- Not necessarily intuitive for all users
- duplicates info already present in the descriptions

jgm commented:
I think that this is probably a good addition.  One question:  the symbol that has this information on hover renders for me as:
<img width="48" alt="Screen Shot 2022-01-07 at 10 04 06 AM" src="https://user-images.githubusercontent.com/3044/148587161-b22af2de-1b10-4ff5-b9a1-e367d33bc241.png">
Is that intended?

not-my-profile commented:
That means your font lacks the glyph for that unicode symbol, it should render as 
![image](https://user-images.githubusercontent.com/73739153/148587523-8ee6dd6f-85d4-446b-acf8-0800440c816a.png)

The character is: 🗹

jgm commented:
Let's try to find a character that has better font coverage, and that also has a good meaning.

tarleb commented:
The plus-minus sign (U+001B) seems like a good option: ±

jgm commented:
Or what about the question-mark symbol that is now used only on hover?

not-my-profile commented:
>Or what about the question-mark symbol that is now used only on hover?

That is a browser cursor ... I don't think it's available as a glyph.

tarleb commented:
There are also 'information source' ℹ and 'circled information source' 🛈. Not sure how well those are supported by common fonts.

jgm commented:
On my browser the first works but not the second.

jgm commented:
I just remembered that our HTML template links in fontawesome.
So we can include icons using this syntax:
```
<span class="fas fa-bars"></span>
```
These will not depend on fonts the user happens to have installed.
Here are the icons we can use:
https://fontawesome.com/v5.15/icons?d=gallery&p=2

jgm commented:
Might look at question-circle, info-circle.

Second note:  looks like the ONLY thing we use fontawesome for are the "bars" for  a dropdown menu with narrow windows.  It might be worth getting rid of this dependency and using plain unicode characters or svgs, both for the bars and for this.

not-my-profile commented:
> looks like the ONLY thing we use fontawesome for are the "bars" for a dropdown menu with narrow windows. It might be worth getting rid of this dependency

I agree (though I think that's better left for another PR).

I updated my PR and [my demo](https://share.push-f.com/pandoc/MANUAL.html#other-extensions) to use the plus minus sign as suggested by @tarleb. I also changed the lists in the title to use `+` and `-`  as list markers ... I think it looks neat.

jgm commented:
Yes, I agree, that looks good!

not-my-profile commented:
The CSS additions aren't live on pandoc.org for some reason. Also there is no newline between the enabled and disabled lists, like it's added by my lua script?

jgm commented:
I just rebuilt the page; I didn't push the new CSS but I can do that.

jgm commented:
Your script doesn't add an extra newline, but I can change it to do so.

jgm commented:
Oh, I do see the code that is meant to add a newline.  I don't know why this isn't happening.

jgm commented:
It appears that the extra newline is added to the title in the AST, but removed when rendering this to HTML.

This is a subtle issue introduced, I think, by 7a9832166e36f77402d5e0259647e9f5c7ba4e58
Pending a fix, I've just added a nonbreaking space for now.

jgm commented:
Bug is now fixed and the page should render properly.

not-my-profile commented:
thank you!
